tests/test_georaster.py

```diff
@@ -638,7 +649,7 @@ def test_set_ndv(self) -> None:
         assert np.all(ndv_index_2 == ndv_index)
 
         # Check that the number of no data value is correct
-        assert np.count_nonzero(ndv_index.data) == 112088
+        assert np.count_nonzero(ndv_index.data) == 111688
```

Comment:
why did this change?

Reply:
I was hoping that you wouldn't catch that hahaha :clown_face:. All of the other parts of the test work, so I assume that this has to do with the new handling of old nodata values. I can get to the bottom of this before merging if you wish.

Reply:
That would be great ! If we have to re-write the tests to make them pass, then there's no point in tests anymore 😉 At least it would be great to understand why that is.

Reply:
Updated the test and documented it properly in 5008906 (which by the way is the first commit hash that I've seen only started with numbers!).

Basically, the previous nodata functionality was that `set_ndv()` just enlarged the mask with the new nodata values, e.g. the pseudocode below:
```python
>>> r.set_ndv(255)  # This masks, say, 300 values
>>> np.count_nonzero(r.data.mask)
300
>>> r.set_ndv(254)  # This masks, say 100 NEW values
>>> np.count_nonzero(r.data.mask)
400
```
In the new case, the `255` values would be unmasked before masking the new `254` value, so the nodata count after this code would be `100`. The functionality for ndarrays (which will be made redundant in this PR), values are not masked but are set to `nan`. There, the previous behaviour of "enlarging the nodata domain" still happens. But yeah, pure ndarrays will most likely not be supported pretty soon anyway.